Add support for token based auth

cjh1 opened the pull request.

synackd commented:
Thanks for this. I updated the title to reflect that this is for the `pcs` command. Is this ready for testing?

cjh1 commented:
> Thanks for this. I updated the title to reflect that this is for the `pcs` command. Is this ready for testing?

yes

cjh1 commented:
To test you would have to define `PCS_JWKS_URL` in the PCS environment ...

cjh1 commented:
I just added it to `docker-compose.test.ct.yml` and added the `quickstart_internal`  to the networks, so it could reach out the opaal from the quickstart setup.

synackd commented:
Do you have a patch you could share for the testing changes?

cjh1 commented:
Sure, here you go:

```
diff --git a/docker-compose.test.ct.yaml b/docker-compose.test.ct.yaml
index 2b48115..2391123 100644
--- a/docker-compose.test.ct.yaml
+++ b/docker-compose.test.ct.yaml
@@ -1,5 +1,7 @@
 networks:
   pcs:
+  quickstart_internal:
+    external: true
 
 services:
 
@@ -318,6 +320,7 @@ services:
       - ETCD_HOST=etcd
       - ETCD_PORT=2379
       - PCS_POWER_SAMPLE_INTERVAL=5
+      - PCS_JWKS_URL=http://opaal:3333/keys
     depends_on:
       - etcd # needed to bring up PCS
       - smd # needed to bring up PCS
@@ -326,6 +329,9 @@ services:
       - wait-for-smd # needed to give PCS time for its initial hardware scan
     networks:
       - pcs
+      - quickstart_internal
+    ports:
+      - 28007:28007
 
   #
   # Tests
@@ -337,18 +343,18 @@ services:
     networks:
       - pcs
 
-  tavern:
-    build:
-      context: test/ct/
-      dockerfile: Dockerfile
-    entrypoint: entrypoint.sh tavern -c /src/app/tavern_global_config_ct_test_emulated_hardware.yaml -p /src/app/api
-    networks:
-      - pcs
+  # tavern:
+  #   build:
+  #     context: test/ct/
+  #     dockerfile: Dockerfile
+  #   entrypoint: entrypoint.sh tavern -c /src/app/tavern_global_config_ct_test_emulated_hardware.yaml -p /src/app/api
+  #   networks:
+  #     - pcs
 
-  smoke:
-    build:
-      context: test/ct/
-      dockerfile: Dockerfile
-    entrypoint: entrypoint.sh smoke -f /src/app/smoke.json -u http://power-control:28007
-    networks:
-      - pcs
+  # smoke:
+  #   build:
+  #     context: test/ct/
+  #     dockerfile: Dockerfile
+  #   entrypoint: entrypoint.sh smoke -f /src/app/smoke.json -u http://power-control:28007
+  #   networks:
+  #     - pcs
```

And you will need the opaal service from the quickstart running ( hence adding quickstart_internal network to be able to reach it ).

synackd reviewed (approved):
LGTM! 🚀
